### ISSUE TITLE
multiple gccX build problems on 10.14

### ISSUE THREAD

**Issue by nieder** [fink/fink-distributions#359]
On 10.14, when Xcode CLI 10.1 is installed, it's possible to not have /usr/include. This leads to the error seen here on gcc5, but I see it also with gcc7:
https://sourceforge.net/p/fink/mailman/message/36582179/
````shell
if [ -d ../prev-gcc ]; then \
          cd ../prev-gcc && \
          /Applications/Xcode.app/Contents/Developer/usr/bin/make real-install-headers-tar DESTDIR=`pwd`/../gcc/ \
            libsubdir=. ; \
        else \
          set -e; for ml in `cat fixinc_list`; do \
            sysroot_headers_suffix=`echo ${ml} | sed -e 's/;.*$//'`; \
            multi_dir=`echo ${ml} | sed -e 's/^[^;]*;//'`; \
            fix_dir=include-fixed${multi_dir}; \
            if ! false && test ! -d `echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`; then \
              echo The directory that should contain system headers does not exist: >&2 ; \
              echo "  `echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`" >&2 ; \
              tooldir_sysinc=`echo "/sw/lib/gcc5/lib/gcc/x86_64-apple-darwin18.2.0/5.5.0/../../../../x86_64-apple-darwin18.2.0/sys-include" | sed -e :a -e "s,[^/]*/\.\.\/,," -e ta`; \
              if test "x`echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`" = "x${tooldir_sysinc}"; \
              then sleep 1; else exit 1; fi; \
            fi; \
            /bin/sh ../../gcc-5.5.0/gcc/../mkinstalldirs ${fix_dir}; \
            chmod a+rx ${fix_dir} || true; \
            (TARGET_MACHINE='x86_64-apple-darwin18.2.0'; srcdir=`cd ../../gcc-5.5.0/gcc; ${PWDCMD-pwd}`; \
              SHELL='/bin/sh'; MACRO_LIST=`${PWDCMD-pwd}`/macro_list ; \
              gcc_dir=`${PWDCMD-pwd}` ; \
              export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
              cd ../build-x86_64-apple-darwin18.2.0/fixincludes && \
              /bin/sh ./fixinc.sh "${gcc_dir}/${fix_dir}" \
                `echo /usr/include | sed -e :a -e 's,[^/]*/\.\.\/,,' -e ta`  ); \
            rm -f ${fix_dir}/syslimits.h; \
            if [ -f ${fix_dir}/limits.h ]; then \
              mv ${fix_dir}/limits.h ${fix_dir}/syslimits.h; \
            else \
              cp ../../gcc-5.5.0/gcc/gsyslimits.h ${fix_dir}/syslimits.h; \
            fi; \
            chmod a+r ${fix_dir}/syslimits.h; \
          done; \
        fi
The directory that should contain system headers does not exist:
  /usr/include
make[3]: *** [stmp-fixinc] Error 1
````

There's a check there for `/usr/include` and so it fails when absent, even though headers could be found in the SDK.

I was able to get around the check by patching gcc/Makefile.in to look for the $SDK/usr/include path instead:
````Patch
--- a/gcc/Makefile.in	2019-02-25 04:19:09.000000000 -0600
+++ b/gcc/Makefile.in	2019-02-25 05:13:32.000000000 -0600
-	    if ! $(inhibit_libc) && test ! -d ${SYSTEM_HEADER_DIR}; then \
+	    if ! $(inhibit_libc) && test ! -d @SDK_PATH@${SYSTEM_HEADER_DIR}; then \
````

Running here, the build then fails right after configuring stage 1 in x86_64-apple-darwin18.2.0/libgcc:
````
/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./gcc/xgcc -B/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./gcc/ -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/bin/ -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/lib/ -isystem /sw/lib/gcc7/x86_64-apple-darwin18.2.0/include -isystem /sw/lib/gcc7/x86_64-apple-darwin18.2.0/sys-include    -g -O2 -O2  -g -O2 -DIN_GCC    -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-format -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition  -isystem ./include   -mmacosx-version-min=10.5 -pipe -fno-common -g -DIN_LIBGCC2 -fbuilding-libgcc -fno-stack-protector   -mmacosx-version-min=10.5 -pipe -fno-common -I. -I. -I../.././gcc -I../../../gcc-7.3.0/libgcc -I../../../gcc-7.3.0/libgcc/. -I../../../gcc-7.3.0/libgcc/../gcc -I../../../gcc-7.3.0/libgcc/../include  -DHAVE_CC_TLS -DUSE_EMUTLS -o _gcov_merge_add.o -MT _gcov_merge_add.o -MD -MP -MF _gcov_merge_add.dep -DL_gcov_merge_add -c ../../../gcc-7.3.0/libgcc/libgcov-merge.c
# If this is the top-level multilib, build all the other
# multilibs.
/bin/sh ../../../../gcc-7.3.0/libgcc/../mkinstalldirs ../../.././gcc/i386
mkdir ../../.././gcc/i386
for file in libgcc_s.1.dylib  libgcc_ext.10.4.dylib libgcc_ext.10.5.dylib; do			\
	  rm -f ../../.././gcc/i386/$file;		\
	  ln -s ../$file ../../.././gcc/i386/;	\
	done
In file included from ../../../gcc-7.3.0/libgcc/libgcov.h:42:0,
                 from ../../../gcc-7.3.0/libgcc/libgcov-merge.c:26:
../../../gcc-7.3.0/libgcc/../gcc/tsystem.h:87:10: fatal error: stdio.h: No such file or directory
 #include <stdio.h>
````
This is solved by adding `-isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk` to CFLAGS`.

But then the build later fails with undefined i386 symbols when linking i386/libgcc_s.dylib:
````
/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./gcc/xgcc -B/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./gcc/ -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/bin/ -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/lib/ -isystem /sw/lib/gcc7/x86_64-apple-darwin18.2.0/include -isystem /sw/lib/gcc7/x86_64-apple-darwin18.2.0/sys-include    -O2  -g -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -DIN_GCC    -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-format -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition  -isystem ./include   -mmacosx-version-min=10.5 -pipe -fno-common -g -DIN_LIBGCC2 -fbuilding-libgcc -fno-stack-protector  -dynamiclib -nodefaultlibs -install_name /sw/lib/gcc7/lib/libgcc_s.1.dylib -single_module -o i386/libgcc_s.dylib -Wl,-exported_symbols_list,libgcc.map -compatibility_version 1 -current_version 1.0 -g -O2 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -m32 -B./ _muldi3_s.o _negdi2_s.o _lshrdi3_s.o _ashldi3_s.o _ashrdi3_s.o _cmpdi2_s.o _ucmpdi2_s.o _clear_cache_s.o _trampoline_s.o __main_s.o _absvsi2_s.o _absvdi2_s.o _addvsi3_s.o _addvdi3_s.o _subvsi3_s.o _subvdi3_s.o _mulvsi3_s.o _mulvdi3_s.o _negvsi2_s.o _negvdi2_s.o _ctors_s.o _ffssi2_s.o _ffsdi2_s.o _clz_s.o _clzsi2_s.o _clzdi2_s.o _ctzsi2_s.o _ctzdi2_s.o _popcount_tab_s.o _popcountsi2_s.o _popcountdi2_s.o _paritysi2_s.o _paritydi2_s.o _powisf2_s.o _powidf2_s.o _powixf2_s.o _powitf2_s.o _mulhc3_s.o _mulsc3_s.o _muldc3_s.o _mulxc3_s.o _multc3_s.o _divhc3_s.o _divsc3_s.o _divdc3_s.o _divxc3_s.o _divtc3_s.o _bswapsi2_s.o _bswapdi2_s.o _clrsbsi2_s.o _clrsbdi2_s.o _fixunssfsi_s.o _fixunsdfsi_s.o _fixunsxfsi_s.o _fixsfdi_s.o _fixdfdi_s.o _fixxfdi_s.o _fixunssfdi_s.o _fixunsdfdi_s.o _fixunsxfdi_s.o _floatdisf_s.o _floatdidf_s.o _floatdixf_s.o _floatundisf_s.o _floatundidf_s.o _floatundixf_s.o _fixsfti_s.o _fixdfti_s.o _fixxfti_s.o _fixtfti_s.o _fixunssfti_s.o _fixunsdfti_s.o _fixunsxfti_s.o _fixunstfti_s.o _floattisf_s.o _floattidf_s.o _floattixf_s.o _floattitf_s.o _floatuntisf_s.o _floatuntidf_s.o _floatuntixf_s.o _floatuntitf_s.o _divdi3_s.o _moddi3_s.o _divmoddi4_s.o _udivdi3_s.o _umoddi3_s.o _udivmoddi4_s.o _udiv_w_sdiv_s.o darwin-64_s.o cpuinfo_s.o tf-signs_s.o sfp-exceptions_s.o addtf3_s.o divtf3_s.o eqtf2_s.o getf2_s.o letf2_s.o multf3_s.o negtf2_s.o subtf3_s.o unordtf2_s.o fixtfsi_s.o fixunstfsi_s.o floatsitf_s.o floatunsitf_s.o fixtfdi_s.o fixunstfdi_s.o floatditf_s.o floatunditf_s.o extendsftf2_s.o extenddftf2_s.o extendxftf2_s.o trunctfsf2_s.o trunctfdf2_s.o trunctfxf2_s.o enable-execute-stack_s.o unwind-dw2_s.o unwind-dw2-fde-darwin_s.o unwind-sjlj_s.o unwind-c_s.o emutls_s.o libgcc.a -lc
ld: warning: The i386 architecture is deprecated for macOS (remove from the Xcode build setting: ARCHS)
ld: warning: could not create compact unwind for __Unwind_RaiseException: does not use EBP or ESP based frame
ld: warning: could not create compact unwind for __Unwind_ForcedUnwind: does not use EBP or ESP based frame
ld: warning: could not create compact unwind for __Unwind_Resume: does not use EBP or ESP based frame
ld: warning: could not create compact unwind for __Unwind_Resume_or_Rethrow: does not use EBP or ESP based frame
ld: warning: ignoring file /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/lib/libc.tbd, missing required architecture i386 in file /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/lib/libc.tbd
Undefined symbols for architecture i386:
  "__keymgr_get_and_lock_processwide_ptr", referenced from:
      _live_image_destructor in unwind-dw2-fde-darwin_s.o
      __Unwind_Find_FDE in unwind-dw2-fde-darwin_s.o
  "__keymgr_set_and_unlock_processwide_ptr", referenced from:
      _live_image_destructor in unwind-dw2-fde-darwin_s.o
      __Unwind_Find_FDE in unwind-dw2-fde-darwin_s.o
  "__keymgr_unlock_processwide_ptr", referenced from:
      __Unwind_Find_FDE in unwind-dw2-fde-darwin_s.o
  "_abort", referenced from:
      ___absvsi2 in _absvsi2_s.o
      ___absvdi2 in _absvdi2_s.o
      ___addvsi3 in _addvsi3_s.o
...many more
````

This is fixed by adding `--disable-multilib` to %c, but the build now fails with missing <stdlib.h>:
````
libtool: compile:  /sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./gcc/xgcc -shared-libgcc -B/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./gcc -nostdinc++ -L/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/x86_64-apple-darwin18.2.0/libstdc++-v3/src -L/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/x86_64-apple-darwin18.2.0/libstdc++-v3/src/.libs -L/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/x86_64-apple-darwin18.2.0/libstdc++-v3/libsupc++/.libs -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/bin/ -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/lib/ -isystem /sw/lib/gcc7/x86_64-apple-darwin18.2.0/include -isystem /sw/lib/gcc7/x86_64-apple-darwin18.2.0/sys-include -I/sw/build.build/gcc7-7.3.0-1.1/gcc-7.3.0/libstdc++-v3/../libgcc -I/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/x86_64-apple-darwin18.2.0/libstdc++-v3/include/x86_64-apple-darwin18.2.0 -I/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/x86_64-apple-darwin18.2.0/libstdc++-v3/include -I/sw/build.build/gcc7-7.3.0-1.1/gcc-7.3.0/libstdc++-v3/libsupc++ -I/sw/include -I/sw/include -D_GLIBCXX_SHARED -fno-implicit-templates -Wall -Wextra -Wwrite-strings -Wcast-qual -Wabi -fdiagnostics-show-location=once -fvisibility-inlines-hidden -ffunction-sections -fdata-sections -frandom-seed=atexit_thread.lo -g -O2 -std=gnu++11 -c ../../../../gcc-7.3.0/libstdc++-v3/libsupc++/atexit_thread.cc  -fno-common -DPIC -D_GLIBCXX_SHARED -o atexit_thread.o
In file included from ../../../../gcc-7.3.0/libstdc++-v3/libsupc++/atexit_thread.cc:25:0:
/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/x86_64-apple-darwin18.2.0/libstdc++-v3/include/cstdlib:75:15: fatal error: stdlib.h: No such file or directory
 #include_next <stdlib.h>
               ^~~~~~~~~~
compilation terminated.
make[5]: *** [atexit_thread.lo] Error 1
````
This is fixed by adding -isysroot to CXXFLAGS. After that, the build finishes stage 1 but fails configuring stage 2 for libbacktrace and I haven't been able to parse the error yet.

**Comment by nieder** [comment]
Current gcc7.info changed that get me this far:
````diff
diff --git a/10.9-libcxx/stable/main/finkinfo/languages/gcc7.info b/10.9-libcxx/stable/main/finkinfo/languages/gcc7.info
index 464064ee4a..8df513f08b 100644
--- a/10.9-libcxx/stable/main/finkinfo/languages/gcc7.info
+++ b/10.9-libcxx/stable/main/finkinfo/languages/gcc7.info
@@ -1,16 +1,20 @@
 Info2: <<
 Package: gcc7
 Version: 7.3.0
-Revision: 1
+Revision: 1.1
 Source: ftp://gcc.gnu.org/pub/gcc/releases/gcc-%v/gcc-%v.tar.xz
 Source-MD5: be2da21680f27624f3a87055c4ba5af2
 PatchFile: %n.patch
 PatchFile-MD5: 07a7b64b31864ff2344c42104fd48d0e
 PatchFile2: %n-apfs.patch
-PatchFile2-MD5: 6f281ae553506c142114c06bf48eb269 
+PatchFile2-MD5: 6f281ae553506c142114c06bf48eb269
+PatchFile3: %n-sdkpath.patch
+PatchFile3-MD5: 0ea945fe1c7d5ea22b1e0dfc3e3fef6b
 PatchScript: <<
 	#!/bin/sh -ev
 	%{default_script}
+	SDK_PATH="$(xcrun --sdk macosx --show-sdk-path)"
+	perl -pi -e "s|\@SDK_PATH\@|$SDK_PATH|g" gcc/Makefile.in
 	# don't update live infodoc dir
 	find . -name Makefile.in | xargs perl -pi -e 's/install-info --version/false/g'
 <<
@@ -64,7 +68,8 @@ ConfigureParams: <<
 	--with-isl=%p \
 	--with-mpc=%p \
 	--with-system-zlib \
-	--program-suffix=-fsf-7
+	--program-suffix=-fsf-7 \
+	--disable-multilib
 <<
 InfoTest: <<
 	TestDepends: <<
@@ -105,6 +110,9 @@ CompileScript: <<
 	ulimit -s `ulimit -s`
 	mkdir ../darwin_objdir
 	cd ../darwin_objdir
+	SDK_PATH="$(xcrun --sdk macosx --show-sdk-path)"
+	export CFLAGS="-isysroot $SDK_PATH"
+	export CXXFLAGS="-isysroot $SDK_PATH"
 	# check for incompatible clang
 	clangvers=`clang --version | cut -d\  -f4`
 	$(`dpkg --compare-versions $clangvers '>>' 6.0`) && $(`dpkg --compare-versions $clangvers '<<' 7.2`) && clangcheck='y'
````

**Comment by nieder** [comment]
The failure seems to be from not finding system headers during stage 2 configure.
````
configure:6461:  /sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./prev-gcc/xg++ -B/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/./prev-gcc/ -B/sw/lib/gcc7/x86_64-apple-darwin18.2.0/bin/ -nostdinc++ -B/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/prev-x86_64-apple-darwin18.2.0/libstdc++-v3/src/.libs -B/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/prev-x86_64-apple-darwin18.2.0/libstdc++-v3/libsupc++/.libs  -I/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/prev-x86_64-apple-darwin18.2.0/libstdc++-v3/include/x86_64-apple-darwin18.2.0  -I/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/prev-x86_64-apple-darwin18.2.0/libstdc++-v3/include  -I/sw/build.build/gcc7-7.3.0-1.1/gcc-7.3.0/libstdc++-v3/libsupc++ -L/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/prev-x86_64-apple-darwin18.2.0/libstdc++-v3/src/.libs -L/sw/build.build/gcc7-7.3.0-1.1/darwin_objdir/prev-x86_64-apple-darwin18.2.0/libstdc++-v3/libsupc++/.libs -c -g -O2   -gtoggle  conftest.cpp >&5
conftest.cpp:21:10: fatal error: stdio.h: No such file or directory
 #include <stdio.h>
          ^~~~~~~~~
compilation terminated.
````
So local flag changes that are used in stage 1 are lost in stage 2? Finding where stages are regulated will suck.
@jwhowarth

**Comment by nieder** [comment]
I found the flags `--with-build-sysroot=$SDK_PATH` and `--with-native-system-header-dir=$SDK_PATH/usr/include`, but using them in any combination causes some different build failures.

`--with-build-sysroot=$SDK_PATH --disable-multilib` and patching the /usr/include check let's the build get the furthest into configuring stage2::gcc, but fails unable to once again find <stdio.h>

relevant: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79885

**Comment by jwhowarth** [comment]
I wouldn't count on this getting properly fixed in the near future. Iain
Sandoe is a big fan of supporting local customized tool chains and not
depending overly on Apple's Xcode. Hence his resistance to using the clang
approach of running the compiler through xrcun in order to set the SDKpath
which clang leverages to find the SDK's /usr/include, etc.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87243

If I recall correctly, the relevant code resides in

https://github.com/llvm-mirror/clang/blob/master/lib/Driver/ToolChains/Darwin.cpp
https://github.com/llvm-mirror/clang/blob/master/lib/Driver/ToolChains/Darwin.h
https://github.com/llvm-mirror/clang/blob/master/lib/Driver/DarwinSDKInfo.cpp

**Comment by jwhowarth** [comment]
If you look at the MacPort gcc8 Portfile, you can see the current mechanism
being used for building FSF gcc on 10.14.

https://github.com/macports/macports-ports/blob/master/lang/gcc8/Portfile

Since they have build separate binary packages for each Darwin release on
MacPorts, the are able to use...

if {${configure.sdkroot} ne ""} {
# We should be using --with-build-sysroot here. Using --with-sysroot
# changes the behavior of the installed gcc to look in that sysroot
# by default instead of /. Using --with-build-sysroot is supposed
# to be used during the build but not impact the installed product.
# Unfortunately, the build fails because the value doesn't get
# plumbed everywhere it is supposed to.
#
# https://trac.macports.org/ticket/53726
# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79885
configure.args-append --with-sysroot="${configure.sdkroot}"
}
which, for their build machines, sets the SDKROOT to a hardcoded
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk.

For fink, you would likely want to change MacOSX10.14.sdk to MacOSX.sdk so
that the symlink is always used rather than a specific SDK version.

Also, since the compilers no longer will link 32-bit code when the SDK is
set (which disables their linkage by masking the 32-bit symbols in
libSystem.dylib through the libSystem.tbd file), you will also want to
pass --disable-multilib
and drop the 32-bit packaging lines in the info file.

**Comment by nieder** [comment]
````
New package: dists/stable/main/binary-darwin-x86_64/languages/gcc7-shlibs_7.3.0-1.2_darwin-x86_64.deb
New package: dists/stable/main/binary-darwin-x86_64/languages/gcc7_7.3.0-1.2_darwin-x86_64.deb
New package: dists/stable/main/binary-darwin-x86_64/languages/gcc7-compiler_7.3.0-1.2_darwin-x86_64.deb
````

**Comment by nieder** [comment]
Allright. And it even works for compiling other packages so that they find $SDK_PATH/usr/include w/out having to specify their own -isysroot at build time.

**Comment by nieder** [comment]
OK, now that gcc7 is building and can compile, need to figure out packaging.
@dmacks 
The patches should only be needed for 10.14 (gcc7 builds fine as is on my 10.13 system, but I only have Xcode 9.4).
10.14 needs to kill the 32bit libraries inside the gcc7 package.
Do I apply the changes to all dists, and too bad if anyone was using gcc7 to locally compile 32bit programs?
Do I conditional the changes to only apply to 10.14?
If so, does this need a completely different gcc7.info file?

**Comment by jwhowarth** [comment]
You will also need to inventory the prior last supported Xcode release for
all supported macOS versions in the current 10.9-libcxx tree to make sure
that they all have the expected symlink at /Applications/Xcode.
app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.
sdk. That will tell you how many releases can be supported by the current
packaging. It probably is simpler to make the new packaging the new default
and to make deployment target variants for the older macOS releases that
don't have the required symlink/

As for the dropping of the 32-bit support, I would probably do that across
the board. Nothing in fink is using it directly and the project to barely
alive so it is hardly worth getting anal retentive about such issues these
days.

**Comment by dmacks** [comment]
I have no objection to killing 32bit for all if that's easiest packaging, though obviously I'd like to keep what we already have (%r separation for 10.14?) if possible.

**Comment by dmacks** [comment]
I was going to ask if we are close to killing old gccX to reduce the amount of hacking (and general support of complex builds) but it looks like there are still a lot of gcc5 users and I vaguely recall some might be difficult to migrate to newer.

**Comment by nieder** [comment]
I'm currently going with 10.14 specific .info files. Minimizes the changes, and reduces the need for conditionals.

**Comment by jwhowarth** [comment]
If you can wait until April, an alternative approach would be to just make
the new gcc9 packaging 32-bit free and start migrating all of the fink
packaging that use that.

**Comment by nieder** [comment]
@jwhowarth : yes, gcc9 should definitely be 64bit only across all dists. There's no reason to keep 32bit code around. For now, the current submission in #361 is a minimal change case.

**Comment by nieder** [comment]
closed by commit 950b823
